mesonbuild/compilers/nim.py

```diff
@@ -0,0 +1,78 @@
+import os.path
+import subprocess
+from ..mesonlib import EnvironmentException
+
+from .compilers import (
+    nim_buildtype_args,
+    Compiler
+)
+
+
+class NimCompiler(Compiler):
+
+    def __init__(self, exelist, version):
+        self.language = 'nim'
+        super().__init__(exelist, version)
+        self.id = 'nim'
+        self.is_cross = False
+
+    def sanity_check(self, work_dir, environment):
+        source_name = os.path.join(work_dir, 'sanity.nim')
```

Comment:
__[Flake8]__

[F841] local variable 'source_name' is assigned to but never used


<!-- Comment by [Sider](https://sider.review) -->